docs/remote-interface.md

```diff
@@ -0,0 +1,67 @@
+# Remote interface
+## Introduction
+
+```
+ +-----------------------------------------------+
+ |               remote interface                |
+ |    +----------------------------------------+ | 
+ |   +----------------------------------------+| |
+ |  +----------------------------------------+|+ |
+ |  |     RabbitMQ-provided RPC              |+  |
+ |  +----------------------------------------+   |
+ |            |                     ^            |
+ +------------|---------------------|------------+
+ +------------|---------------------|------------+
+ |            |   RabbitMQ broker   |            |
+ |            |                     |            |
+ |            |             +------------------+ |
+ |            v            +------------------+| |
+ |  +------------------+  +------------------+|| |
+ |  | incoming queue   |  | private queue    ||| |
+ |  |                  |  |                  ||| |
+ |  | - trytes         |  | - PoW result     ||| |
+ |  | - mwm            |  |                  ||+ |
+ |  |                  |  |                  |+  |
+ |  +------------------+  +------------------+   |
+ |            |                     ^            |
+ +------------|---------------------|------------+
+              v                     |
+   +---------------------------------------------+
+  +---------------------------------------------+|
+ +---------------------------------------------+|+
+ |                  remote worker              |+
+ +---------------------------------------------+
+```
+To support asynchronouse remote procedure call, remote interface, `Remote_ImplContext`, in dcurl provides an interface to implement it. dcurl currently uses RabbitMQ C client to implement asynchronouse RPC in remote interface. Remote interface provides thread management to can run an asynchronouse RPC per thread.

```

Comment:
Typo `asynchronous`.
And it should be
```
To support asynchronous remote procedure call and remote interface, `Remote_ImplContext` in dcurl ......
```

`Remote interface provides thread management to can run an asynchronouse`?
Another typo and the `to can run` seems weird.